File: CS/DocumentServerExample/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraRichEdit;
using System.IO;
using System.Diagnostics;

namespace DocumentServerExample
{
    public partial class Form1 : Form
    {
        List<string> fileList = new List<string>();
        Dictionary<DocumentFormat, string> dictF = new Dictionary<DocumentFormat, string>();
        RichEditDocumentServer server;
        PerformanceCounter counter;

        public Form1()
        {
            InitializeComponent();
            InitializeFormatDictionary();
            server = new RichEditDocumentServer();
            
            textBox1.Text = Application.StartupPath;
            timer1.Tick += new EventHandler(timer1_Tick);

            string procName = Process.GetCurrentProcess().ProcessName;
            counter = new PerformanceCounter("Process", "Working Set - Private", procName);
            ShowMemoryUsage();
        }

        private void InitializeFormatDictionary()
        {
            dictF.Add(DocumentFormat.OpenXml, "docx");
            dictF.Add(DocumentFormat.Rtf, "rtf");
            dictF.Add(DocumentFormat.PlainText, "txt");
            dictF.Add(DocumentFormat.Doc, "doc");
            dictF.Add(DocumentFormat.ePub, "epub");
            dictF.Add(DocumentFormat.OpenDocument, "odt");
            dictF.Add(DocumentFormat.WordML, "xml");
            dictF.Add(DocumentFormat.Html, "htm");
            dictF.Add(DocumentFormat.Mht, "mht");
        }

        private void PerformConversion(string filePath, DocumentFormat sourceFormat, DocumentFormat destFormat)
        {
            FileStream fsIn = File.OpenRead(filePath);
            string outFileName = Path.ChangeExtension(filePath,dictF[destFormat]);
            server.LoadDocument(fsIn,sourceFormat);
            FileStream fsOut = File.Open(outFileName, FileMode.Create);

            if (destFormat == DocumentFormat.Rtf) {
                server.Options.Export.Rtf.Compatibility.DuplicateObjectAsMetafile = false;
            }
   
            server.SaveDocument(fsOut,destFormat);
            fsIn.Close();
            fsOut.Close();
        }

        private void UpdateFileList(string path)
        {
            if (Directory.Exists(path)) {
                string[] files = System.IO.Directory.GetFiles(path, "*.docx", System.IO.SearchOption.AllDirectories);

                foreach (string file in files) {
                    if (!fileList.Contains(file)) {
                        fileList.Add(file);
                        PerformConversion(file, DocumentFormat.OpenXml, DocumentFormat.Rtf);
                    }
                }
            }
        }

        private void btnConvert_Click(object sender, EventArgs e)
        {
            if (timer1.Enabled) {
                timer1.Stop();
                btnConvert.Text = "Start!";
            }
            else {
                timer1.Start();
                btnConvert.Text = "Stop!";
            }
        }

        void timer1_Tick(object sender, EventArgs e)
        {
            UpdateFileList(textBox1.Text);
            ShowMemoryUsage();
        }
        private void ShowMemoryUsage()
        {
            label1.Text = String.Format("Memory usage: {0:N0} K", counter.RawValue/1024);
        }
    }
}

File: VB/DocumentServerExample/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraRichEdit
Imports System.IO
Imports System.Diagnostics

Namespace DocumentServerExample

    Public Partial Class Form1
        Inherits Form

        Private fileList As List(Of String) = New List(Of String)()

        Private dictF As Dictionary(Of DocumentFormat, String) = New Dictionary(Of DocumentFormat, String)()

        Private server As RichEditDocumentServer

        Private counter As PerformanceCounter

        Public Sub New()
            InitializeComponent()
            InitializeFormatDictionary()
            server = New RichEditDocumentServer()
            textBox1.Text = Application.StartupPath
            AddHandler timer1.Tick, New EventHandler(AddressOf timer1_Tick)
            Dim procName As String = Process.GetCurrentProcess().ProcessName
            counter = New PerformanceCounter("Process", "Working Set - Private", procName)
            ShowMemoryUsage()
        End Sub

        Private Sub InitializeFormatDictionary()
            dictF.Add(DocumentFormat.OpenXml, "docx")
            dictF.Add(DocumentFormat.Rtf, "rtf")
            dictF.Add(DocumentFormat.PlainText, "txt")
            dictF.Add(DocumentFormat.Doc, "doc")
            dictF.Add(DocumentFormat.ePub, "epub")
            dictF.Add(DocumentFormat.OpenDocument, "odt")
            dictF.Add(DocumentFormat.WordML, "xml")
            dictF.Add(DocumentFormat.Html, "htm")
            dictF.Add(DocumentFormat.Mht, "mht")
        End Sub

        Private Sub PerformConversion(ByVal filePath As String, ByVal sourceFormat As DocumentFormat, ByVal destFormat As DocumentFormat)
            Dim fsIn As FileStream = File.OpenRead(filePath)
            Dim outFileName As String = Path.ChangeExtension(filePath, dictF(destFormat))
            server.LoadDocument(fsIn, sourceFormat)
            Dim fsOut As FileStream = File.Open(outFileName, FileMode.Create)
            If destFormat = DocumentFormat.Rtf Then
                server.Options.Export.Rtf.Compatibility.DuplicateObjectAsMetafile = False
            End If

            server.SaveDocument(fsOut, destFormat)
            fsIn.Close()
            fsOut.Close()
        End Sub

        Private Sub UpdateFileList(ByVal path As String)
            If Directory.Exists(path) Then
                Dim files As String() = Directory.GetFiles(path, "*.docx", SearchOption.AllDirectories)
                For Each file As String In files
                    If Not fileList.Contains(file) Then
                        fileList.Add(file)
                        PerformConversion(file, DocumentFormat.OpenXml, DocumentFormat.Rtf)
                    End If
                Next
            End If
        End Sub

        Private Sub btnConvert_Click(ByVal sender As Object, ByVal e As EventArgs)
            If timer1.Enabled Then
                timer1.Stop()
                btnConvert.Text = "Start!"
            Else
                timer1.Start()
                btnConvert.Text = "Stop!"
            End If
        End Sub

        Private Sub timer1_Tick(ByVal sender As Object, ByVal e As EventArgs)
            UpdateFileList(textBox1.Text)
            ShowMemoryUsage()
        End Sub

        Private Sub ShowMemoryUsage()
            label1.Text = String.Format("Memory usage: {0:N0} K", counter.RawValue / 1024)
        End Sub
    End Class
End Namespace
